parquet/converted_types.py

```diff
@@ -1,155 +1,127 @@
-if hasattr(int, 'from_bytes'):
-    b2int = lambda x: int.from_bytes(x, 'big')
-else:
-    import codecs
-    b2int = lambda x: int(codecs.encode(x, 'hex'), 16)
+from decimal import Decimal
 
+from .thrift_structures import parquet_thrift
+logger = logging.getLogger('parquet')  # pylint: disable=invalid-name
 
-def invert_dict(d):
-    "Map key:value to value:key"
-    return {v:k for k, v in d.items()}
+try:
+    from bson import BSON
+except ImportError:
+    def BSON(x):
+        raise ImportError("BSON not found")
 
+PY3 = sys.version_info.major > 2
 
-def map_spark_timestamp(x):
-    """Special conversion for 'timestamp' column as created by spark, and
-    possibly hive. Such a column does not have a 'converted type' defined,
-    but is instead labeled in the alternative metadata stored in the footer
-    key-value aread.
-    
-    Data should be a column/series of 12-byte values (INT96).
-    
-    Use with series.map(map_spark_timestamp)
+# define bytes->int for non 2, 4, 8 byte ints
+if PY3:
+    def intbig(data):
+        """Convert big ints using python 3's built-in support."""
+        return int.from_bytes(data, 'big', signed=True)
+else:
+    def intbig(data):
+        """Convert big ints using a hack of encoding bytes as hex and decoding to int."""
+        return int(codecs.encode(data, 'hex'), 16)
+
+DAYS_TO_MILLIS = 86400000000000
+"""Number of millis in a day. Used to convert a Date to a date"""
+
+
+def typemap(se):
+    """Get the final (pandas) dtype - no actual conversion"""
+    if se.converted_type is None:
+        tconv =  {parquet_thrift.Type.INT32: np.int32,
+                  parquet_thrift.Type.INT64: np.int64,
+                  parquet_thrift.Type.FLOAT: np.float32,
+                  parquet_thrift.Type.DOUBLE: np.float64,
+                  parquet_thrift.Type.BOOLEAN: np.bool_,
+                  parquet_thrift.Type.INT96: np.dtype('S12'),
+                  parquet_thrift.Type.BYTE_ARRAY: np.dtype("O")}
+        if se.type in tconv:
+            return tconv[se.type]
+        else:
+            return np.dtype("S%i" % se.type_length)
+    tconv = {parquet_thrift.ConvertedType.UTF8: np.dtype("O"),
+             parquet_thrift.ConvertedType.DECIMAL: np.float64,
+             parquet_thrift.ConvertedType.UINT_8: np.uint8,
+             parquet_thrift.ConvertedType.UINT_16: np.uint16,
+             parquet_thrift.ConvertedType.UINT_32: np.uint32,
+             parquet_thrift.ConvertedType.UINT_64: np.uint64,
+             parquet_thrift.ConvertedType.INT_8: np.int8,
+             parquet_thrift.ConvertedType.INT_16: np.int16,
+             parquet_thrift.ConvertedType.INT_32: np.int32,
+             parquet_thrift.ConvertedType.INT_64: np.int64,
+             parquet_thrift.ConvertedType.TIME_MILLIS: np.dtype('<m8[ns]'),
+             parquet_thrift.ConvertedType.DATE: np.dtype('<M8[ns]'),
+             parquet_thrift.ConvertedType.TIMESTAMP_MILLIS: np.dtype('<M8[ns]')
+             }
```

Comment:
These dictionaries seem generally useful.  I recommend pulling them out of the function definition so that they become accessible to others down the line who need this same data.